### ISSUE TITLE
Error: Failed to set ICE candidate; RTCPeerConnection is closed.

### ISSUE THREAD

**Issue by t-mullen** [node-webrtc/node-webrtc#498]
This error is thrown when attempting to renegotiate after adding a new track to the initiator.

It only occurs when there is a datachannel and ice is trickling. The correct ontrack event is still fired and the peer connection continues to be usable (is NOT closed). I don't see this behaviour on any browsers.

Here's the simplest reproduction case I could get:
```javascript
const wrtc = require('wrtc')
(async function () {
  const config = {
    iceServers: [
      {
        urls: 'stun:stun.l.google.com:19302'
      },
      {
        urls: 'stun:global.stun.twilio.com:3478?transport=udp'
      }
    ],
    sdpSemantics: 'unified-plan'
  }
  const peer1 = new wrtc.RTCPeerConnection(config)
  const peer2 = new wrtc.RTCPeerConnection(config)

  const dc = peer1.createDataChannel('name', {})
  dc.onopen = () => {
    setInterval(() => {
      dc.send('ping') // prove the pc is NOT closed
    }, 1000)
  }

  peer2.ondatachannel = (event) => {
    event.channel.onmessage = (event) => {
      console.log(event.data)
    }
  }

  // the error seems to occur more frequently when these handlers are registered...
  peer1.onsignalingstatechange = () => {
    console.log(peer1.signalingState)
  }
  peer2.onsignalingstatechange = () => {
    console.log(peer2.signalingState)
  }

  peer1.ontrack = (event) => {
    console.log('1 got track')
  }
  peer2.ontrack = (event) => {
    console.log('2 got track')
  }

  peer1.onicecandidate = (event) => {
    if (!event.candidate) return
    peer2.addIceCandidate(event.candidate).catch(err => {
      console.log(err, peer1.connectionState, peer1.signalingState)
    })
  }
  peer2.onicecandidate = (event) => {
    if (!event.candidate) return
    peer1.addIceCandidate(event.candidate).catch(err => {
      console.log(err, peer1.connectionState, peer1.signalingState)
    })
  }

  // negotiate
  var offer = await peer1.createOffer()
  await peer1.setLocalDescription(offer)
  await peer2.setRemoteDescription(offer)
  var answer = await peer2.createAnswer()
  await peer2.setLocalDescription(answer)
  await peer1.setRemoteDescription(answer)

  // add a new track on initiator
  var source = new wrtc.nonstandard.RTCVideoSource()
  var track = source.createTrack()
  var stream = new wrtc.MediaStream()
  peer1.addTrack(track, stream)
  
  // wait, then renegotiate
  setTimeout(async function () {
    var offer = await peer1.createOffer()
    await peer1.setLocalDescription(offer)
    await peer2.setRemoteDescription(offer)
    var answer = await peer2.createAnswer()
    peer1.setRemoteDescription(answer)
  }, 4000)
})()
```

**Comment by t-mullen** [comment]
I was able to work around this by catching the error and checking if `pc.signalingState` is actually closed.

It seems the method here of getting the value for `pc.signalingState` returns "closed" incorrectly?
https://github.com/node-webrtc/node-webrtc/blob/dd208f65534ef05d416ec91edf9648352be9469a/src/peerconnection.cc#L515

**Comment by markandrus** [comment]
Thanks a lot for the report and the repro case, @t-mullen! There’s definitely an issue with the `!=` comparison in the C++ code, but I’ll also try to identify why adding that candidate fails to begin with.

**Comment by markandrus** [comment]
This is released in v0.3.7
